### ISSUE TITLE
Java 12 to 19 and Java 21 have Java 20 as dependency

### ISSUE THREAD

**Issue by omginput** [OpenMandrivaAssociation/distribution#2894]
**OpenMandriva version**:
ROME, on Rock Java 12 to Java 19 except 17 depend on Java 17

**Describe the bug**:
Java 12 to 19 and Java 21 bring Java 20 as dependency although the shouldn't

**Steps to reproduce**:
Try to install Java 12 to 19 and Java 21 and you'll see

**Observed behavior**:
Java 12 to 19 and Java 21 bring Java 20 as dependency although the shouldn't

**Expected behavior**:
Java 12 to 19 and Java 21 don't bring Java 20 as dependency

**Additional comment**:
Crucial to get fixed

**Logs and screenshots if relevant**
![image](https://user-images.githubusercontent.com/8403565/231519321-92ad3dda-2f2a-4ecd-bfe3-8d07f9e1d90a.png)

**Comment by benbullard79** [comment]
I *think* this can be confirmed.

**Comment by ZeroAbility** [comment]
The spec for `java-24-openjdk appears to have the correct `Requires` set. Is this still an issue?

**Comment by josiahwelch** [comment]
I went through `java-12-openjdk` thru `java-19-openjdk` and `java-21-openjdk` and `java-20-openjdk` is no longer listed as a dependency for them. However `java-12-openjdk-devel` thru `java-16-openjdk-devel` and they were listing `java-21-openjdk` and `java-21-openjdk-devel` as dependencies... I am assuming that's a bug but could be wrong.

**Comment by benbullard79** [comment]
I don't think this is a bug. I heard it called bootstrapping but am not sure what bootstrapping means in this context.
